Add documentation for service discovery (#150)
eclipse-chariott/chariott: service_discovery/samples/README.md

# Samples

- [Introduction](#introduction)
- [Simple Discovery Sample](#simple-discovery-sample)
  - [Provider](#provider)
  - [Consumer](#consumer)

## Introduction

The samples show applications that can be used with the Service Registry. For any sample, make sure that you first have service discovery running:

```shell
cargo run -p service_discovery
```

## Simple Discovery Sample

> This example shows a how a provider and consumer communicate, using the service registry to register and discover. The two applicaitons then directly communicate with one another, through a known interface. You can add the same logic to your own applications to get started using Service Discovery.

Reviewer: Please change this to "This example shows how a ... "

> This example shows a how a provider and consumer communicate, using the service registry to register and discover. The two applicaitons then directly communicate with one another, through a known interface. You can add the same logic to your own applications to get started using Service Discovery.

Reviewer 2: ```suggestion
This example shows a how a provider and consumer communicate, using the service registry to register and discover. The two applications then directly communicate with one another, through a known interface. You can add the same logic to your own applications to get started using Service Discovery.
```

> The provider application consists of one gRPC service, the Hello World service. This service just logs and returns a message with "Hello, " followed by the input string from the request. The provider registers itself with the service registry, so that it can be consumed by other applications. The consumer "discovers" the location of the hello world service by calling discover on the service registry. It validates that the communication_kind (which can include the protocol and api specification type) and communication_reference (a string to identify the api specification) are what it expects and knows how to communicate with. The consumer then calls the SayHello method on the uri provided in the discover response to use the provider directly. You will see the output of these operations in the terminal windows when you run the services.

Reviewer: Please consider having the provider and consumer as different paragraphs.

> The provider application consists of one gRPC service, the Hello World service. This service just logs and returns a message with "Hello, " followed by the input string from the request. The provider registers itself with the service registry, so that it can be consumed by other applications. The consumer "discovers" the location of the hello world service by calling discover on the service registry. It validates that the communication_kind (which can include the protocol and api specification type) and communication_reference (a string to identify the api specification) are what it expects and knows how to communicate with. The consumer then calls the SayHello method on the uri provided in the discover response to use the provider directly. You will see the output of these operations in the terminal windows when you run the services.

Reviewer: Please capitalize the Hello World service in "The consumer "discover" the location of the hello world service ..."

> The provider application consists of one gRPC service, the Hello World service. This service just logs and returns a message with "Hello, " followed by the input string from the request. The provider registers itself with the service registry, so that it can be consumed by other applications. The consumer "discovers" the location of the hello world service by calling discover on the service registry. It validates that the communication_kind (which can include the protocol and api specification type) and communication_reference (a string to identify the api specification) are what it expects and knows how to communicate with. The consumer then calls the SayHello method on the uri provided in the discover response to use the provider directly. You will see the output of these operations in the terminal windows when you run the services.

Reviewer: Please specify the subject in this sentence "It validates that the ... are what it expects ... ". This may confuse the reader if they don't know what "it" refers to.

### Provider

> To run the provider, from the simple discovery provider directory in a different terminal window run:

Reviewer 2: This wording doesn't feel quite right to me, maybe something like. Okay leaving if you dont want to change:

```suggestion
To run the provider, open a new terminal window under the sample discovery provider directory and run:
```

```shell
cargo run
```

### Consumer

> To run the consumer, from the simple discovery consumer directory in a different terminal window run:

Reviewer 2: See provider comment for rewording suggestion
